### DISCUSSION TITLE
Device in rack layout redirects to IP address

### DISCUSSION THREAD

**Discussion by rottenteeth** [netbox-community/netbox#13836] [Help Wanted!]
NetBox version: 3.5.9

Open rack layout and click to device. Hyperlinks of devices refer to IP address of server rather than the domain. 
https://XXX.XXX.XXX.XXX/dcim/devices/X/ rather than https://domain.com/dcim/devices/X/

**Comment by candlerb** [comment]
You've not configured your front-end proxy (apache or nginx) correctly. It needs either to preserve the incoming Host header or add X-Forwarded-Host. There are working sample configs [here](https://github.com/netbox-community/netbox/tree/master/contrib).

See #13404, #7912, #6209.

**Comment by rottenteeth** [comment]
Ok, I got it.

In my case it is nginx.

/etc/nginx/sites-available netbox.conf

Instead 
proxy_set_header X-Forwarded-Host $http_host;

I used
proxy_set_header X-Forwarded-Proto http;

Thanks.

**Reply by candlerb** [reply]
"X-Forwarded-Proto" tells the server behind the proxy whether the user connected using http or https. The values are "http" or "https" only.

"X-Forwarded-Host" tells the server behind the proxy what hostname the user originally connected to.

You need to set both, as the sample config does. For Netbox to be able to assemble an absolute URL like `https://domain.com/dcim/devices/X/`, it needs to know which scheme to use (https) and which hostname to use (domain.com)

It also sets `X-Real-IP` (whereas Apache sets [X-Forwarded-For](https://httpd.apache.org/docs/2.4/mod/mod_proxy.html#x-headers)). This is for when Netbox know the client IP address, e.g. for logging. It won't affect the URLs being generated though.
